### ISSUE TITLE
Feature: Update the develop branch for deployment on heroku.

### ISSUE THREAD

**Issue by codesankalp** [anitab-org/open-source-programs-backend#80]
### Is your feature request related to a problem? Please describe.
The final phase of any project is the successful deployment so that users can interact with it in production.
This website backend is currently under development but it will be beneficial for newcomers or any contributors to see the website in the production and know about this project in detail.

### Describe the solution you'd like
1. Use the develop branch as a deployment branch.
2. Do the required changes to deploy the site on Heroku.
3. You can see this for reference: [heroku docs](https://devcenter.heroku.com/articles/deploying-python) or [this](https://www.analyticsvidhya.com/blog/2020/10/step-by-step-guide-for-deploying-a-django-application-using-heroku-for-free/)
4. The deployment must be integrated with GitHub.
**NOTE**: Allow the `open-source-programs-web` deployed URL to make API requests by adding the deployed URL to allowed hosts.

### Additional context
Feel free to research more about this and add more features to make the deployment process easy.

### Estimation
2 hours

**Comment by codesankalp** [comment]
@isabelcosta Can you add labels to this issue?

**Comment by isabelcosta** [comment]
@codesankalp do we need an additional branch for this? on mentorship-backend project I am using the `develop` branch for heroku deployment.

**Comment by codesankalp** [comment]
@isabelcosta We can deploy using the develop branch also. Should I edit the title and description of this issue?

**Comment by Rahulm2310** [comment]
@isabelcosta @codesankalp I would like to work on this.

**Comment by codesankalp** [comment]
@Rahulm2310 Please wait until @isabelcosta will clarify the branch to use for deployment.

**Comment by isabelcosta** [comment]
yes, from develop branch so that every time something is merged the deployment is updated @codesankalp

**Comment by codesankalp** [comment]
@Rahulm2310 Assigning you !!
@isabelcosta Can you add the labels?

**Comment by Rahulm2310** [comment]
@codesankalp do I just need to add the settings (like procfile, add django-heroku) and submit a PR ?
Also, how are we going to deploy it to heroku as we have already used up the heroku free limit.

**Comment by codesankalp** [comment]
@Rahulm2310 Make the required changes needed to deploy to Heroku using a single command. You can use any methods about this. We will discuss changes in the PR.
I don't know about the Heroku free limit of Anitab-org, as in the session it was decided that we are going to host this backend on Heroku.

@isabelcosta Is Heroku free limit has been used by the organization? If yes, then what we need to use?

**Comment by isabelcosta** [comment]
@codesankalp we are not yet using Heroku as an organization (though we intend to). For now, all you need @Rahulm2310 is to add whatever is needed to the code that allows us to deploy this on heroku with single clicks. So for mentorship-backend it was only adding `.procfile`, so you can do this, and if more is needed also add to the PR :)

**Comment by Rahulm2310** [comment]
@isabelcosta @codesankalp should I add deploy instructions in README.md ?

**Comment by codesankalp** [comment]
I think adding deployment instruction is not required in README as this instruction is not required by contributor.
@Rahulm2310

**Comment by isabelcosta** [comment]
@Rahulm2310 you can add it in the PR description so that we can do it. you could also add instructions on deploying to Heroku, on a `docs` folder as a markdown file ;) then it would go to docusaurus

**Comment by Rahulm2310** [comment]
@isabelcosta the PR for migrating to docusaurus #99 is under review. I can add the file in `docs` folder only after it gets merged. For now should I add the instructions to the PR description?

**Comment by Rahulm2310** [comment]
@isabelcosta the PR for migrating to docusaurus #99 is under review. I can add the file in `docs` folder only after it gets merged. For now should I add the instructions to the PR description?
